content/en/docs/tasks/job/intra-job-pod-networking-using-pod-hostnames.md

```diff
@@ -0,0 +1,108 @@
+
+To enable pod-to-pod communication using pod hostnames, you must do the following:
+
+1. Set up a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
+with a valid label selector for the pods created by your job. This will trigger the 
+Kubernetes' service discovery mechanism to cache the hostnames of 
+the pods running your job (note that the service discovery mechanism does not need to be 
+in the same namepsace as the job pods). One easy way to do this is to use the `job-name: <your-job-name>`
+selector, since the job-name label will be automatically added by Kubernetes. 
+
+**Note**: if you are using MiniKube or a similar tool, you may need to take [extra steps](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/) to ensure you have DNS.
+
+2. Update the template spec in your job with the following: `subdomain: <headless-svc-name>`
+   where `<headless-svc-name>` must match the name of your headless service
+   exactly. 
+
+Example job which completes only after all pods successfully ping each other using hostnames:
+
+```yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-svc
+spec:
+  clusterIP: None # clusterIP must be None to create a headless service
+  selector:
+    job-name: example-job # must match job template spec label
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: example-job
+spec:
+  completions: 3
+  parallelism: 3
+  completionMode: Indexed
+  template:
+    spec:
+      subdomain: headless-svc # has to match headless service name
+      restartPolicy: Never
+      containers:
+      - name: example-workload
+        image: bash:latest
+        command:
+        - bash
+        - -c
+        - |
+          for i in 0 1 2
+          do
+            gotStatus="-1"
+            wantStatus="0"             
+            while [ $gotStatus -ne $wantStatus ]
+            do                                       
+              ping -c 1 example-job-${i}.headless-svc > /dev/null 2>&1
+              gotStatus=$?                
+              if [ $gotStatus -ne $wantStatus ]; then
+                echo "Failed to ping pod example-job-${i}.headless-svc, retrying in 1 second..."
+                sleep 1
+              fi
+            done                                                         
+            echo "Successfully pinged pod: example-job-${i}.headless-svc"
+          done
+```
+
+After applying the example above, pods will be able to reach each other over the network
+using: `<pod-hostname>.<headless-service-name>`. You should see output similar to the following:
+```
+$ kubectl logs example-job-0-qws42
+Failed to ping pod example-job-0.headless-svc, retrying in 1 second...
+Successfully pinged pod: example-job-0.headless-svc
+Successfully pinged pod: example-job-1.headless-svc
+Successfully pinged pod: example-job-2.headless-svc
+```
+**NOTE**: It is important note that the `<pod-hostname>.<headless-service-name>` name format used
```

Comment:
You can use `{{< note > }}` and `{{< /note >}}`

Reply:
Done